### DISCUSSION TITLE
Problem retrieving publications by author started 7/8/2021

### DISCUSSION THREAD

**Discussion by BISCoder** [scholarly-python-package/scholarly#299] [Q&A] [answered]
Hello I am new to GitHub and hope I follow protocol posting an issue and a discussion. Respect intended!

My program was processing fine until morning 7/8/2021.

Then Scholarly started giving me the error: <class 'KeyError'> on calls such as resp = scholarly.fill(q), where q is an author object newly returned from a search_author_id(q) where q was a GoogleScholarID as shown (zzz) in a web page https://scholar.google.com/citations?user=zzz&hl=en
This had been working wonderfully before.

I tried calling fill(author, sections=['basics', 'indices', 'coauthors', 'counts'] that was fine but [] and 'publications' both failed.

Anyone else having this concern? Or else advice?

**Comment by KenrickM** [comment]
I had the same problem and edited the code in site-packages/scholarly/publication_parser.py   
on line 11   from 'data-href'   to just 'href'  and that seemed to work for me.

I also had problems filling a publication object, around line 269 of the same file, couldn't find gsc_vcd but could find gsc_oci?

**Comment by BISCoder** [comment, marked as answer]
Hello - I think the team (ipeirotis) fixed this issue. I am feeling kind of happy because I had never helped out on an open source project before and I figured out part of the fix and deployed it my downloaded package and passed on what I learned.  Of course ipeirotis put a more elegant solution in place. Much appreciated.

**Reply by Aniruth-ai** [reply]
Hi @BISCoder,

I don't know, but kind of still facing the same issue. I have updated the scholarly package to the latest version. The code below used to run perfectly fine:

search_query = scholarly.search_author(author)
author_publications = next(search_query).fill(sections = ['publications'])

but now, it throws an error **AttributeError: 'dict' object has no attribute 'fill'**

Seems like you resolved the issue, could you be able to help me on this?

P.S. My entire application revolves around getting publications w.r.t the Author. So if this seems to be not working for me, my entire application is useless.

Thank you!

**Reply by ipeirotis** [reply]
Please give the complete, executable, piece of code that causes the issue (including the name of the author, so that we can run the code and check what is wrong)

**Reply by Aniruth-ai** [reply]
Hi @ipeirotis,

**_Code Snippet_**:
from scholarly import scholarly
search_query = scholarly.search_author('M Nikdast')
author = next(search_query).fill(sections = ['publications'])
print(author)

**_Error_**:
author = next(search_query).fill(sections = ['publications'])
AttributeError: 'dict' object has no attribute 'fill'

Thank You.

**Reply by ipeirotis** [reply]
```python
from scholarly import scholarly
search_query = scholarly.search_author('M Nikdast')
author = scholarly.fill(next(search_query), sections = ['publications'])
scholarly.pprint(author)
```

**Reply by ipeirotis** [reply]
You use an incorrect pattern for the `fill` command (you seem to use the code from the versions 0.x )

**Reply by Aniruth-ai** [reply]
My bad, I dont know how I missed that.

Thanks to you @ipeirotis. You saved my application.
